### PR TITLE
UX and code cleanup work

### PR DESCRIPTION
**Pull request by freeqaz**
Specifically:
- Standardized the front-end on using camelCase (refactored snake_case)
- Cleaned up a bunch of styles in the app
- Added more loading states
- Added additional text content to help the user in various places to better explain the app
- Removed anonymous functions on the front-end and replaced with named functions to provide for better stack traces
- Some additional linting rules for sanity (you can use console statements in dev mode now, thank god)

Note: There are a bunch of whitespace changes here because I de-indented the files I was working on. Otherwise my IDE kept putting spaces in the wrong place and I wanted to die. I will run a full linter run in another commit, so as to leave this code actually reviewable!

### CODE REVIEW

File: front-end/vue.config.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 const webpack = require('webpack')
 	
 module.exports = {
-  publicPath: './',
+  publicPath: '/',
```
Comment (mandatoryprogrammer) [inline front-end/vue.config.js]: This **might** break prod if I'm reading it right, what was the reason for this change?

File: front-end/src/main.js [hunk 2]
```diff
@@ -8,44 +8,72 @@ import {faFontAwesome} from '@fortawesome/free-brands-svg-icons'
...
+            }
+        ]
+    }
+];
 
 const router = new VueRouter({
-    routes // short for `routes: routes`
-})
+    routes, // short for `routes: routes`
+    base: '/',
```
Comment (mandatoryprogrammer) [inline front-end/src/main.js]: This *might* break prod if I'm reading it right, what was the reason for this change?

**Review by mandatoryprogrammer — CHANGES_REQUESTED** [review]
Some questions to answer but mostly looks good.

File: front-end/vue.config.js (hunk 1 above)
Comment (freeqaz) [reply front-end/vue.config.js]: Nice catch!

I see what you mean. This is required for Dev to work w/ history mode router. If you go to a nested page like `/layer/:uuid` then it tries to pull assets from `/layer/public/etc` and breaks.

But in prod we have everything living behind a proxied route, which is a special case. I can make sure to resolve this by having a production-specific configuration.

**Comment by freeqaz** [comment]
I'm going to close out this PR in favor of the SSR one. I fixed the bug with the public path over there.

**Pull request closed without merging**